base/compiler/abstractinterpretation.jl

```diff
@@ -901,6 +901,9 @@ function const_prop_entry_heuristic(interp::AbstractInterpreter, result::MethodC
         # thus there won't be much benefit in constant-prop' here
         add_remark!(interp, sv, "[constprop] Disabled by entry heuristic (limited accuracy)")
         return false
+    elseif result.edge_effects.nothrow !== ALWAYS_TRUE
+        # Could be improved to Bottom (or at least could see the effects improved)
+        return true
     else
         add_remark!(interp, sv, "[constprop] Disabled by entry heuristic (unimprovable return type)")
         return false
```

Comment:
Maybe we want to enable this only for `Const` because `PartialTypeVar`/`PartialOpaque` usually mean we already have interesting enough information?
```suggestion
    else
        if isa(rt, Const)
            if result.edge_effects.nothrow !== ALWAYS_TRUE
                # Could still be improved to Bottom (or at least could see the effects improved)
                return true
            end
        end
        add_remark!(interp, sv, "[constprop] Disabled by entry heuristic (unimprovable result)")
        return false
```